io/tools/convert_clouds_meshes.cpp

```diff
@@ -0,0 +1,311 @@
+      output_type = BINARY_COMPRESSED;
+    else
+    {
+      PCL_ERROR("Wrong output type!\n");
+      displayHelp (argc, argv);
+      return (-1);
+    }
+  }
+
+  // Find input/output file names
+  std::vector<int> file_args;
+  for (int i = 0; i < obj_files.size (); ++i)
+    file_args.push_back (obj_files[i]);
+  for (int i = 0; i < pcd_files.size (); ++i)
+    file_args.push_back (pcd_files[i]);
+  for (int i = 0; i < ply_files.size (); ++i)
+    file_args.push_back (ply_files[i]);
+  for (int i = 0; i < stl_files.size (); ++i)
+    file_args.push_back (stl_files[i]);
+  for (int i = 0; i < vtk_files.size (); ++i)
+    file_args.push_back (vtk_files[i]);
+  // Re-order, the first argument is always the input file
+  std::sort (file_args.begin (), file_args.end ());
+  if (file_args.size () != 2)
+  {
+    PCL_ERROR("Wrong input/output file count!\n");
+    displayHelp (argc, argv);
+    return (-1);
+  }
+
+  // If the input is a PCD file it can't be a mesh: treat it as a point cloud
+  if (boost::filesystem::path (argv[file_args[0]]).extension () == ".pcd")
+  {
+    pcl::PCLPointCloud2::Ptr cloud (new pcl::PCLPointCloud2);
+    //Eigen::Vector4f origin; // TODO: Support origin/orientation
+    //Eigen::Quaternionf orientation;
+    if (pcl::io::loadPCDFile (argv[file_args[0]], *cloud) < 0)  //, origin, orientation) < 0)
+    {
+      PCL_ERROR("Unable to load %s.\n", argv[file_args[0]]);
+      return (-1);
+    }
+
+    PCL_INFO("Loaded a point cloud with %d points (total size is %d) and the following channels:\n%s\n",
+             cloud->width * cloud->height, cloud->data.size (), pcl::getFieldsList (*cloud).c_str ());
+
+    if (!savePointCloud (cloud, argv[file_args[1]], output_type))
+    {
+      PCL_ERROR("Failed to save %s.\n", argv[file_args[0]]);
+      return (-1);
+    }
+  }
+
+  // Try to load as mesh
+  pcl::PolygonMesh mesh;
+  if (pcl::io::loadPolygonFile (argv[file_args[0]], mesh) != 0)
+  {
+    PCL_INFO("Loaded a point cloud with %d points (total size is %d) and the following channels:\n%s\n",
+             mesh.cloud.width * mesh.cloud.height, mesh.cloud.data.size (), pcl::getFieldsList (mesh.cloud).c_str ());
+
+    if (saveMesh (mesh, argv[file_args[1]], output_type))
+      return (-1);
+  }
+  else if (boost::filesystem::path (argv[file_args[0]]).extension () != ".stl")  // STL files cannot be point clouds
+  {
+    // OBJ, PLY or VTK
+    PCL_WARN("Could not load %s as a mesh, trying as a point cloud instead.\n", argv[file_args[1]]);
+    pcl::PCLPointCloud2::Ptr cloud (new pcl::PCLPointCloud2);
+
+    PCL_INFO("Loaded a point cloud with %d points (total size is %d) and the following channels:\n%s\n",
+             cloud->width * cloud->height, cloud->data.size (), pcl::getFieldsList (*cloud).c_str ());
+
+    if (!savePointCloud (cloud, argv[file_args[1]], output_type))
+    {
+      PCL_ERROR("Failed to save %s.\n", argv[file_args[0]]);
+      return (-1);
+    }
+  }
```

Comment:
In case your .stl file is corrupted and cannot be loaded as a mesh, the program will exit normally.